<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Assessing Sequencing Data Quality (FASTQC)  | Intro to Sequencing Data &amp; UNIX Tutorial</title>
  <meta name="description" content="Chapter 5 Assessing Sequencing Data Quality (FASTQC)  | Intro to Sequencing Data &amp; UNIX Tutorial" />
  <meta name="generator" content="bookdown 0.46 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Assessing Sequencing Data Quality (FASTQC)  | Intro to Sequencing Data &amp; UNIX Tutorial" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Assessing Sequencing Data Quality (FASTQC)  | Intro to Sequencing Data &amp; UNIX Tutorial" />
  
  
  

<meta name="author" content="Christina Costa" />


<meta name="date" content="2026-02-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="save-your-terminal-session.html"/>
<link rel="next" href="submit-your-work.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Week 3</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="basic-commands-and-view-files.html"><a href="basic-commands-and-view-files.html"><i class="fa fa-check"></i><b>1</b> <span style="color: green;"> Basic Commands and View Files </span></a></li>
<li class="chapter" data-level="2" data-path="extract-information-about-our-sequence-data.html"><a href="extract-information-about-our-sequence-data.html"><i class="fa fa-check"></i><b>2</b> <span style="color: green;"> Extract Information about our Sequence Data </span></a></li>
<li class="chapter" data-level="3" data-path="fastq-sample-info.html"><a href="fastq-sample-info.html"><i class="fa fa-check"></i><b>3</b> <span style="color: green;"> FASTQ Sample Info </span></a></li>
<li class="chapter" data-level="4" data-path="save-your-terminal-session.html"><a href="save-your-terminal-session.html"><i class="fa fa-check"></i><b>4</b> <span style="color: green;"> Save Your Terminal Session </span></a></li>
<li class="chapter" data-level="5" data-path="assessing-sequencing-data-quality-fastqc.html"><a href="assessing-sequencing-data-quality-fastqc.html"><i class="fa fa-check"></i><b>5</b> <span style="color: green;"> Assessing Sequencing Data Quality (FASTQC) </span></a></li>
<li class="chapter" data-level="6" data-path="submit-your-work.html"><a href="submit-your-work.html"><i class="fa fa-check"></i><b>6</b> <span style="color: green;"> Submit Your Work </span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Intro to Sequencing Data &amp; UNIX Tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="assessing-sequencing-data-quality-fastqc" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> <span style="color: green;"> Assessing Sequencing Data Quality (FASTQC) </span><a href="assessing-sequencing-data-quality-fastqc.html#assessing-sequencing-data-quality-fastqc" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Download and open the FASTQC results (two HTML files) for both the files you viewed above.</p>
<div id="question-3-fastq-summary.-fastqc-will-also-show-you-some-summary-stats.-from-the-report-for-each-fastq-file-what-is-the-total-number-of-sequences-the-total-number-of-bases-what-is-the-range-in-sequence-length" class="section level4 hasAnchor" number="5.0.0.1">
<h4><span class="header-section-number">5.0.0.1</span> <span style="color: blue;"> <strong>Question 3</strong> (FASTQ Summary). FASTQC will also show you some summary stats. From the report, for each FASTQ file, what is the <strong>total number of sequences</strong>? The <strong>total number of bases</strong>? What is the <strong>range in sequence length</strong>? </span><a href="assessing-sequencing-data-quality-fastqc.html#question-3-fastq-summary.-fastqc-will-also-show-you-some-summary-stats.-from-the-report-for-each-fastq-file-what-is-the-total-number-of-sequences-the-total-number-of-bases-what-is-the-range-in-sequence-length" class="anchor-section" aria-label="Anchor link to header"></a></h4>
</div>
<div id="question-4.-sequencing-quality.-do-you-notice-any-quality-issues-in-the-fastqc-report-of-either-file-what-might-each-indicate-about-the-data" class="section level4 hasAnchor" number="5.0.0.2">
<h4><span class="header-section-number">5.0.0.2</span> <span style="color: blue;"> <strong>Question 4</strong>. (Sequencing Quality). Do you notice any quality issues in the FASTQC report of either file? What might each indicate about the data? </span><a href="assessing-sequencing-data-quality-fastqc.html#question-4.-sequencing-quality.-do-you-notice-any-quality-issues-in-the-fastqc-report-of-either-file-what-might-each-indicate-about-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Review the info on each parameter in more depth here:
<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/" class="uri">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/</a></p>
<p>Take a look at a “good Illumina FASTQC report”:
<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html" class="uri">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html</a></p>
<p>And a bad:
<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html" class="uri">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html</a></p>
<details>
<summary>
Click to view more about the ancient sample
</summary>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="assessing-sequencing-data-quality-fastqc.html#cb18-1" tabindex="-1"></a><span class="st">&quot;The ancient DNA actually appears to be fairly decent quality from this quick report, besides what are clearly some flow cell failures. Although it is ancient, it is not nearly as old as many samples we have extracted DNA from. In addition, this is a modern publication, meaning they were using the most up to date library prep methods for ancient DNA, which likely increased the quality. The one thing you may have noticed is a shorter fragment length for some that is not just do to shorter reads but the input DNA being &quot;</span></span></code></pre></div>
</div>
<div id="question-5.-sequencing-technology-compare-the-illumina-err12040069---short-read-and-pacbio-hg02486---long-read-fastqc-reports.-what-differences-do-you-notice" class="section level4 hasAnchor" number="5.0.0.3">
<h4><span class="header-section-number">5.0.0.3</span> <span style="color: blue;"> <strong>Question 5</strong>. (Sequencing Technology): Compare the Illumina (ERR12040069 - Short read) and PacBio (HG02486 - Long read) FASTQC reports. What differences do you notice? </span><a href="assessing-sequencing-data-quality-fastqc.html#question-5.-sequencing-technology-compare-the-illumina-err12040069---short-read-and-pacbio-hg02486---long-read-fastqc-reports.-what-differences-do-you-notice" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Most of the time a few orange or even red symbols does not mean something is terribly wrong and you need to throw out your data.</strong> Some of these errors can be addressed during data processing (Week 6), e.g., trimming low quality bases especially at the ends of reads, removing adapters if present in reads (although our sequences here have none!).</p>
<p>FASTQC was created for short-read Illumina sequencing data, and therefore is not well-tuned for other types of data.</p>
<p>For example, you likely noticed the odd presence of <strong>PolyA</strong> content in the <strong>Adapter Content</strong> summary for the long reads. This is a known artifact. A Poly-A tail as a reminder is a repetitive sequence of <strong>A</strong>s that is added after transcription to the 3’ end of a messenger RNA transcript. Remember, we are sequencing the genomic DNA here, not RNA, so these reads should not actually contain real Poly-A sequences.</p>
<p>I have never done long-read sequencing but I believe e.g., <strong>PacBIO SMRT</strong> has there own quality control report they provide that is more specific to the data generated.</p>
<p>Another program <strong>fastp</strong> claims to work better on long read data. Download the <strong>fastp.HG02486.html</strong> from Brightspace as well and compare this to the FASTQC report. This provides similar information in a different format.</p>
<p>Note: I generated these FASTQC reports for you to avoid having to install software. But you can run <strong>fastqc</strong> on your sequencing data from the <strong>command line</strong> (or the available App) and it will generate the same output. You can download fastqc here:
(<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" class="uri">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a>).</p>
<p>Feel free to download and do so using the provided FASTQ files. However, we can do the same via a web-based application called Galaxy that we will return to in Week 6, and for our final assignment.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="assessing-sequencing-data-quality-fastqc.html#cb19-1" tabindex="-1"></a><span class="ex">fastqc</span> unknown.fastq</span></code></pre></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="save-your-terminal-session.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="submit-your-work.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/USERNAME/REPO/edit/BRANCH/Sequencing_Unix_Intro.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
