<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Extract Information about our Sequence Data  | Intro to Sequencing Data &amp; UNIX Tutorial</title>
  <meta name="description" content="Chapter 2 Extract Information about our Sequence Data  | Intro to Sequencing Data &amp; UNIX Tutorial" />
  <meta name="generator" content="bookdown 0.46 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Extract Information about our Sequence Data  | Intro to Sequencing Data &amp; UNIX Tutorial" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Extract Information about our Sequence Data  | Intro to Sequencing Data &amp; UNIX Tutorial" />
  
  
  

<meta name="author" content="Christina Costa" />


<meta name="date" content="2026-02-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="basic-commands-and-view-files.html"/>
<link rel="next" href="fastq-sample-info.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Week 3</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="basic-commands-and-view-files.html"><a href="basic-commands-and-view-files.html"><i class="fa fa-check"></i><b>1</b> <span style="color: green;"> Basic Commands and View Files </span></a></li>
<li class="chapter" data-level="2" data-path="extract-information-about-our-sequence-data.html"><a href="extract-information-about-our-sequence-data.html"><i class="fa fa-check"></i><b>2</b> <span style="color: green;"> Extract Information about our Sequence Data </span></a></li>
<li class="chapter" data-level="3" data-path="fastq-sample-info.html"><a href="fastq-sample-info.html"><i class="fa fa-check"></i><b>3</b> <span style="color: green;"> FASTQ Sample Info </span></a></li>
<li class="chapter" data-level="4" data-path="save-your-terminal-session.html"><a href="save-your-terminal-session.html"><i class="fa fa-check"></i><b>4</b> <span style="color: green;"> Save Your Terminal Session </span></a></li>
<li class="chapter" data-level="5" data-path="assessing-sequencing-data-quality-fastqc.html"><a href="assessing-sequencing-data-quality-fastqc.html"><i class="fa fa-check"></i><b>5</b> <span style="color: green;"> Assessing Sequencing Data Quality (FASTQC) </span></a></li>
<li class="chapter" data-level="6" data-path="submit-your-work.html"><a href="submit-your-work.html"><i class="fa fa-check"></i><b>6</b> <span style="color: green;"> Submit Your Work </span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Intro to Sequencing Data &amp; UNIX Tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="extract-information-about-our-sequence-data" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> <span style="color: green;"> Extract Information about our Sequence Data </span><a href="extract-information-about-our-sequence-data.html#extract-information-about-our-sequence-data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>A useful feature of a FASTA file is that each entry starts with the same symbol “&gt;”. If we count this symbol we can know how many sequences are in our file. In this case we of course already know, but for some larger files this may come in handy.</p>
<p>We will use a command called <strong>grep</strong>. (stands for global regular expression print). It searches for text lines in a file that matches what is known as a regular expression. You can get fancy with this, but we do not need to.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="extract-information-about-our-sequence-data.html#cb9-1" tabindex="-1"></a><span class="fu">grep</span> <span class="st">&quot;&gt;&quot;</span> ace2.fa</span></code></pre></div>
<p>Now this prints the lines that have that character in parentheses.</p>
<p>But we want to know how many lines. Thankfully, grep has a flag for that! We are going to add the <strong>-c</strong> flag, which will <strong>count</strong> the number of lines in our file with that symbol. In our case, this should be five, the number of ACE2 sequences we have.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="extract-information-about-our-sequence-data.html#cb10-1" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-c</span> <span class="st">&quot;&gt;&quot;</span> ace2.fa</span></code></pre></div>
<p>Now, download the two FASTQ files from Brightspace for today. Remember this is the raw data we get from the sequencing machine.</p>
<p>One of these contains short reads and the other long reads.</p>
<p>A very useful character to know is the <strong>asterisk</strong> symbol, which is a universal expression, it stands for anything, better known as a <strong>wild card</strong>.</p>
<p>The following line will show you all your files ending in <strong>fastq.gz</strong>. Make sure to change your directory if needed using <strong>cd</strong> to the file location.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="extract-information-about-our-sequence-data.html#cb11-1" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-lht</span> <span class="pp">*</span>fastq.gz</span></code></pre></div>
<p>Note: these files are in a <strong>compressed</strong> format (<strong>.gz</strong>). The same commands will work but first you need to unzip it. Do not do this by clicking on the file! Keep it zipped. We will only unzip it in the Terminal.</p>
<p>Use the <strong>gzcat</strong> command followed by your file name. This is the same as the <strong>cat</strong> command but it works on <strong>gzipped</strong> files.</p>
<p>Then we will introduce a new topic called <strong>piping</strong>. Piping is when you write one command, then insert this character <strong>“|”</strong>. Then you write another command that is performed on whatever the first output is.</p>
<p>Essentially you are feeding the output of the first part into the second command. Remember, you need to type the actual file name:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb12-1"><a href="extract-information-about-our-sequence-data.html#cb12-1" tabindex="-1"></a><span class="ex">gzcat</span> unknown.fastq.gz <span class="kw">|</span> <span class="fu">less</span></span></code></pre></div>
<p>Piping is one type of <strong>input</strong> and <strong>output</strong> redirection. This can be done in several ways:</p>
<ul>
<li>Store program’s output in a file to be saved (<strong>&gt;</strong>).</li>
<li>Direct a program to take input from a file (<strong>&lt;</strong>).</li>
<li>Direct the output to be another program’s input (<strong>|</strong>)</li>
</ul>
<p>Take a look at each using <strong>less</strong>. Do you notice any differences between the FASTA file and these two FASTQ files?</p>
<p>We can count how many lines total are in a file with the <strong>wc -l</strong> command. You can do this for both but it will give you the same answer because I subset these files to the same number of lines (the complete sequencing data files were quite large).</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="extract-information-about-our-sequence-data.html#cb13-1" tabindex="-1"></a><span class="ex">gzcat</span> unknown.fastq.gz <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span> </span></code></pre></div>
<p>But that does not tell us how many <strong>reads</strong> are in a file, which is what we are usually interested in.</p>
<p>To do that we rely on the fact that each <strong>FASTQ file should have 4 lines per read</strong>. So we can simply divide the number we got above by 4 to get the total number of reads. Or we can do the more complicated line below. Note: <strong>bc</strong> stands for basic calculator.</p>
<p>Unfortunately this will not work without the <strong>echo</strong> which prints the value of the command in parentheses following the <strong>$</strong> to standard output so we can then divide by 4. Do not worry too much about this, if you start spending a lot of time doing shell scripting you will get used to some of these features.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb14-1"><a href="extract-information-about-our-sequence-data.html#cb14-1" tabindex="-1"></a><span class="bu">echo</span> <span class="va">$(</span><span class="ex">gzcat</span> unknown.fastq <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span><span class="va">)</span>/4 <span class="kw">|</span> <span class="fu">bc</span> </span></code></pre></div>
<p>In theory we could use the same logic as above for the FASTA and just <strong>grep</strong> and count the number of <strong>“@”</strong> signs that start the FASTQ header. Unfortunately, this same character can be used in the last quality score line (Phred score).</p>
<div id="question-1.-from-only-visually-examining-the-reads-which-of-these-fastq-files-is-most-likely-the-output-of-long-read-sequencing-and-which-is-from-short-read-sequencing" class="section level4 hasAnchor" number="2.0.0.1">
<h4><span class="header-section-number">2.0.0.1</span> <span style="color: blue;"> <strong>Question 1</strong>. From only visually examining the reads, which of these FASTQ files is most likely the output of long-read sequencing and which is from short-read sequencing? </span><a href="extract-information-about-our-sequence-data.html#question-1.-from-only-visually-examining-the-reads-which-of-these-fastq-files-is-most-likely-the-output-of-long-read-sequencing-and-which-is-from-short-read-sequencing" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>In this case we are not interested in the number of reads in the sequencing run but how many bases are sequenced in each read.</p>
<p>There is a way to do this in command line using some fancy <strong>awk</strong> commands but that is beyond the scope of this intro. I will include the line for reference below so you can check your answer.</p>
</div>
<div id="question-2.-do-you-think-these-reads-are-single-or-paired-end-hint-look-at-the-number-of-fastq-files-i-provided-you-for-each-sample-and-the-headers-of-the-reads." class="section level4 hasAnchor" number="2.0.0.2">
<h4><span class="header-section-number">2.0.0.2</span> <span style="color: blue;"> <strong>Question 2</strong>. Do you think these reads are single or paired-end? <em>Hint</em>: look at the number of FASTQ files I provided you for each sample and the headers of the reads. </span><a href="extract-information-about-our-sequence-data.html#question-2.-do-you-think-these-reads-are-single-or-paired-end-hint-look-at-the-number-of-fastq-files-i-provided-you-for-each-sample-and-the-headers-of-the-reads." class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Note: while paired-end reads are useful in short-read technology like Illumina, and help with e.g., aligning to a reference genome, with long-read sequencing it is not needed.</p>
<p>This line will output the average read length in the FASTQ file:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="extract-information-about-our-sequence-data.html#cb15-1" tabindex="-1"></a><span class="ex">gzcat</span> unknown.fastq.gz <span class="kw">|</span> <span class="fu">awk</span> <span class="st">&#39;NR%4==2 {print length}&#39;</span> <span class="kw">|</span> <span class="fu">awk</span> <span class="st">&#39;{sum += $1; count++} END {print sum/count}&#39;</span></span></code></pre></div>
<details>
<summary>
For those interested more info on the command:
</summary>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="extract-information-about-our-sequence-data.html#cb16-1" tabindex="-1"></a><span class="st">&quot;To break this down, NR stands for number of records, or current line number, the % is a remainder symbol (the leftover value when dividing current line number in this case by 4), and with the == 2, is essentially saying for every four line block, print the length of the second line. Remember that every read entry has four lines, and the second line has the actual sequence. </span></span>
<span id="cb16-2"><a href="extract-information-about-our-sequence-data.html#cb16-2" tabindex="-1"></a></span>
<span id="cb16-3"><a href="extract-information-about-our-sequence-data.html#cb16-3" tabindex="-1"></a><span class="st">The length of each sequence line is then piped into another awk command, where the sequence lengths are summed together. count++ keeps a record of how many sequences have been summed.</span></span>
<span id="cb16-4"><a href="extract-information-about-our-sequence-data.html#cb16-4" tabindex="-1"></a></span>
<span id="cb16-5"><a href="extract-information-about-our-sequence-data.html#cb16-5" tabindex="-1"></a><span class="st">Once that is done END, it then prints the sum of all the sequences divided by the count (so that average read length).</span></span>
<span id="cb16-6"><a href="extract-information-about-our-sequence-data.html#cb16-6" tabindex="-1"></a></span>
<span id="cb16-7"><a href="extract-information-about-our-sequence-data.html#cb16-7" tabindex="-1"></a><span class="st">This type of syntax is really quite complex I often have to look it up again and can only do basic things off the top of my head. For now, I am only interested in you being able to use some of the basic command line tools shown earlier. &quot;</span></span></code></pre></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="basic-commands-and-view-files.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="fastq-sample-info.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/USERNAME/REPO/edit/BRANCH/Sequencing_Unix_Intro.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
